engine/access/ingestion/collections/indexer_test.go

```diff
@@ -0,0 +1,432 @@
+		defer cancel()
+
+		ready := make(chan struct{})
+		done := make(chan struct{})
+		go func() {
+			defer close(done)
+			indexer.WorkerLoop(ctx, func() { close(ready) })
+		}()
+
+		synctest.Wait()
+		unittest.RequireClosed(t, ready, "worker should be ready")
+
+		// calling OnCollectionReceived stores and indexes the collection when the collection is not in storage.
+		indexer.OnCollectionReceived(collection)
+		synctest.Wait()
+
+		// calling OnCollectionReceived does skips indexing when the collection is already in storage.
+		s.collections.On("LightByID", collection.ID()).Unset()
+		s.collections.On("LightByID", collection.ID()).Return(collection.Light(), nil).Once()
+
+		indexer.OnCollectionReceived(collection)
+		synctest.Wait()
+	})
+}
+
+func (s *IndexerSuite) TestWorkerProcessing_ProcessesCollections() {
+	g := fixtures.NewGeneratorSuite()
+	rootBlock := g.Blocks().Genesis()
+
+	RunWithBlockchain(s.T(), func(bc *blockchain) {
+		initializer := store.NewConsumerProgress(bc.db, module.ConsumeProgressLastFullBlockHeight)
+		progress, err := initializer.Initialize(rootBlock.Height)
+		s.Require().NoError(err)
+
+		lastFullBlockHeight, err := counters.NewPersistentStrictMonotonicCounter(progress)
+		s.Require().NoError(err)
+
+		indexer, err := NewIndexer(
+			unittest.Logger(),
+			metrics.NewNoopCollector(),
+			bc.state,
+			bc.all.Blocks,
+			bc.collections,
+			lastFullBlockHeight,
+			bc.lockManager,
+		)
+		s.Require().NoError(err)
+
+		cm := component.NewComponentManagerBuilder().
+			AddWorker(indexer.WorkerLoop).
+			Build()
+
+		signalerCtx, cancel := irrecoverable.NewMockSignalerContextWithCancel(s.T(), context.Background())
+		defer cancel()
+
+		cm.Start(signalerCtx)
+		unittest.RequireCloseBefore(s.T(), cm.Ready(), 100*time.Millisecond, "could not start worker")
+		defer func() {
+			cancel()
+			unittest.RequireCloseBefore(s.T(), cm.Done(), 100*time.Millisecond, "could not stop worker")
+		}()
+
+		prevHeader := rootBlock.ToHeader()
+		for range 10 {
+			block, collectionList := s.fixtureWithParent(g, prevHeader)
+			bc.finalizeBlock(s.T(), block)
+			prevHeader = block.ToHeader()
+
+			for _, collection := range collectionList {
+				indexer.OnCollectionReceived(collection)
+			}
+
+			// wait until all collections are indexed
+			require.Eventually(s.T(), func() bool {
+				missingCollections, err := indexer.MissingCollectionsAtHeight(block.Height)
+				return err == nil && len(missingCollections) == 0
+			}, time.Second, 10*time.Millisecond)
+
+			// check that all collection indices are populated
+			// ByID uses all of the indices.
+			for _, collection := range collectionList {
+				actual, err := bc.collections.ByID(collection.ID())
+				s.Require().NoError(err)
+				s.Require().Equal(collection, actual)
+			}
+		}
+
+		// wait until the last full block height is updated
+		require.Eventually(s.T(), func() bool {
+			return lastFullBlockHeight.Value() == prevHeader.Height
+		}, time.Second, 100*time.Millisecond)
```

Comment:
```suggestion
		}, time.Second, 10*time.Millisecond)
```
Not sure, but 100ms seems like a long minimum time to wait.